src/relay/azext_relay/_params.py

```diff
@@ -0,0 +1,85 @@
+    for scope in ['relay namespace show', 'relay namespace delete']:
+        with self.argument_context(scope) as c:
+            c.argument('namespace_name', options_list=['--name', '-n'], help='name of the Namespace')
+
+    # region Namespace Authorizationrule
+    for scope in ['relay namespace authorizationrule', 'relay namespace authorizationrule keys list', 'relay namespace authorizationrule keys renew']:
+        with self.argument_context(scope) as c:
+            c.argument('authorization_rule_name', options_list=['--name', '-n'], help='name of the Namespace AuthorizationRule')
+
+    with self.argument_context('relay namespace authorizationrule create') as c:
+        c.argument('accessrights', options_list=['--access-rights'], arg_type=get_enum_type(['Send', 'Listen', 'Manage']), help='Authorization rule rights of type list')
+
+    with self.argument_context('relay namespace authorizationrule keys renew') as c:
+        c.argument('key_type', options_list=['--key-name'], arg_type=get_enum_type(['PrimaryKey', 'SecondaryKey']), help='specifies Primary or Secondary key needs to be reset')
+
+
+# region - WCF Relay Create
+def load_arguments_wcfrelay(self, _):
+    with self.argument_context('relay wcfrelay') as c:
+        c.argument('relay_name', options_list=['--name', '-n'], help='Name of WCF Relay')
+
+    with self.argument_context('relay wcfrelay create') as c:
+        c.argument('relay_type', options_list=['--relay-type'], arg_type=get_enum_type(['NetTcp', 'Http']), help='WCF relay type.')
+        c.argument('requires_client_authorization', options_list=['--requires-client-authorization'], type=bool, help='True if client authorization is needed for this relay; otherwise, false.')
+        c.argument('requires_transport_security', options_list=['--requires-transport-security'], type=bool, help='True if transport security is needed for this relay; otherwise, false.')
+        c.argument('user_metadata', options_list=['--user-metadata'], help='The usermetadata is a placeholder to store user-defined string data for the WCF Relay endpoint. For example, it can be used to store descriptive data, such as list of teams and their contact information. Also, user-defined configuration settings can be stored.')
+
+    # region WCF Relay Authorizationrule
+    for scope in ['relay wcfrelay authorizationrule', 'relay wcfrelay authorizationrule keys list', 'relay wcfrelay authorizationrule keys renew']:
+        with self.argument_context(scope) as c:
+            c.argument('authorization_rule_name', options_list=['--name', '-n'], help='name of the WCF Relay AuthorizationRule')
+            c.argument('relay_name', options_list=['--wcfrelay-name'], help='name of the WCF Relay')
+
+    with self.argument_context('relay wcfrelay authorizationrule create') as c:
+        c.argument('rights', options_list=['--access-rights'], arg_type=get_enum_type(['Send', 'Listen', 'Manage']), help='AuthorizationRule rights of type list')
```

Comment:
Since you are using this in multiple places, to avoid duplication of maintenance effort, I'd recommend you create an argument type for this:
```Python
rights_arg_type = CLIArgumentType(options_list=[...], arg_type=get_enum_type([...]), help='...')

with self.argument_context('relay authorization-rule ...') as c:
  c.argument('rights', arg_type=rights_arg_type)


with self.argument_context('relay namespace ...') as c:
  c.argument('accessrights', arg_type=rights_arg_type, options_list=['--access-rights'])

Reply:
implemented